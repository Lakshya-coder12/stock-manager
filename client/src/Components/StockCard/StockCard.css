.card {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-radius: 10px;
  height: 100px;
  background-color: white;
  padding: 10px;
  margin-top: 15px;
  box-shadow: 2px 2px #babab8;
}

.arrow {
  position: absolute;
  top: 10px;
  right: 10px;
  cursor: pointer;
  color: #1976c2;
}

.info {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.card-title {
  color: #1976c2;
  font-size: large;
  font-weight: 700;
}

.companyName {
  color: #1976c2;
  font-size: medium;
  font-weight: 600;
}
