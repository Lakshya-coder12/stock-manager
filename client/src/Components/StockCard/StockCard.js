import React, { Component } from "react";
import "./StockCard.css";
import { Link } from "react-router-dom";
import Button from "../Button/Button";
import StockIncrementDecrementModal from "../StockIncrementDecrementModel/index";

class StockCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
    };
  }
  handleClickOpen = () => {
    this.setState({
      isOpen: true,
    });
  };
  handleClose = () => {
    this.setState({
      isOpen: false,
    });
  };
  render() {
    return (
      <div className="card">
        <div className="arrow">
          <Button onClick={this.handleClickOpen}>Inc/Dec Stock</Button>
        </div>
        <div className="card-title">{this.props.name}</div>
        <div className="info">
          <div className="companyName">{`Company: ${this.props.companyName}`}</div>
          <Link to={`/details/${this.props.id}`}>
            <Button>More Details</Button>
          </Link>
        </div>
        <StockIncrementDecrementModal
          stock={this.props.stock}
          id={this.props.id}
          open={this.state.isOpen}
          onClose={this.handleClose}
        />
      </div>
    );
  }
}

export default StockCard;
