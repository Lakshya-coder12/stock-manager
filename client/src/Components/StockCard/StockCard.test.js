import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import StockCard from "./StockCard";

configure({ adapter: new Adapter() });

describe("Testing StockCard component", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<StockCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
