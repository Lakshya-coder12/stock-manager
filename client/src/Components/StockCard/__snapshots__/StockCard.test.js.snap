// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing StockCard component matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="arrow"
  >
    <Button
      onClick={[Function]}
    >
      Inc/Dec Stock
    </Button>
  </div>
  <div
    className="card-title"
  />
  <div
    className="info"
  >
    <div
      className="companyName"
    >
      Company: undefined
    </div>
    <Link
      to="/details/undefined"
    >
      <Button>
        More Details
      </Button>
    </Link>
  </div>
  <Connect(StockIncrementDecrementModal)
    onClose={[Function]}
    open={false}
  />
</div>
`;
